Userland/Libraries/LibWeb/HTML/HTMLFormElement.cpp

```diff
@@ -179,17 +180,33 @@ WebIDL::ExceptionOr<void> HTMLFormElement::submit_form(JS::NonnullGCPtr<HTMLElem
 
     // 11. If method is dialog, then:
     if (method == MethodAttributeState::Dialog) {
-        // FIXME: 1. If form does not have an ancestor dialog element, then return.
-        // FIXME: 2. Let subject be form's nearest ancestor dialog element.
-        // FIXME: 3. Let result be null.
+        Node* ancestor = this;
+        do {
+            ancestor = ancestor->parent_element();
+        } while (ancestor && !is<HTMLDialogElement>(ancestor));
+
+        // 1. If form does not have an ancestor dialog element, then return.
+        if (!ancestor)
+            return {};
+
+        // 2. Let subject be form's nearest ancestor dialog element.
+        auto* subject = static_cast<HTMLDialogElement*>(ancestor);
+
+        // 3. Let result be null.
+        Optional<String> result;
+
         // FIXME: 4. If submitter is an input element whose type attribute is in the Image Button state, then:
         //           1. Let (x, y) be the selected coordinate.
         //           2. Set result to the concatenation of x, ",", and y.
-        // FIXME: 5. Otherwise, if submitter has a value, then set result to that value.
-        // FIXME: 6. Close the dialog subject with result.
-        // FIXME: 7. Return.
 
-        dbgln("FIXME: Implement form submission with `dialog` action. Returning from form submission.");
+        // 5. Otherwise, if submitter has a value, then set result to that value.
+        if (submitter->has_attribute(AttributeNames::value))
+            result = submitter->get_attribute_value(AttributeNames::value);
```

Comment:
I think this can avoid 2 lookups, as our `get_attribute` returns "null" if not found:
```c++
result = submitter->get_attribute(AttributeNames::value);
```